### ISSUE TITLE
proposal: on 64-bit Windows, download 64-bit MinGW & Nim

### ISSUE THREAD

**Issue by akavel** [dom96/choosenim#128]
As far as I understand, choosenim currently [just downloads a 32-bit version of MinGW on Windows](https://github.com/dom96/choosenim/blob/121133ba06b0954e058d38743a18fe1ad1f817f2/src/choosenim.nim#L36) (if no gcc is found in PATH), and then, as a consequence, a 32-bit version of Nim. Is there any interest in allowing choosenim to download a 64-bit version of MinGW (and Nim)?

If yes, should there be a flag for choosing the variant, or should the 64-bit toolchain be the default one on 64-bit Windows ([autodetected](https://blogs.msdn.microsoft.com/david.wang/2006/03/27/howto-detect-process-bitness/))?

**Comment by dom96** [comment]
This is already done AFAIK. Not sure what the rules are exactly, @genotrance will know.

**Comment by genotrance** [comment]
No, this is still open - the 64-bit change was only to install 64-bit Nim if 64-bit gcc was detected. If there is no gcc present, 32-bit is the default.

My proposal to fix this is to automatically download 64-bit if OS supports it. So if CPU = 64, install 64, else install 32. Only if the user states `--32bit` or similar will choosenim revert to 32-bit installs.

Once this is done, we need to then allow 32-bit and 64-bit installs to coexist. So if 32-bit flag is specified, choosenim will have to adjust the version of gcc being pointed to. But that's a separate issue and discussion.

**Comment by chenzhekl** [comment]
@genotrance was right. Choosenim still downloaded the 32 bit mingw and nim for my Windows 10 64bit machine.

**Comment by Skaruts** [comment]
Is there a way to manually trick choosenim into installing 64 bit stuff?
